src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -681,6 +681,10 @@ export class SavedObjectsRepository {
       { ignore: [404] }
     );
 
+    if (isNotFoundFromUnsupportedServer({ statusCode, headers })) {
```

Comment:
Originally, we only checked the product header when `body.result === 'not_found'` and have no way of knowing that the response will actually contain that. Now we check _any_ 404 that's returned.